File: topics/dynamic-programming/easy/0070-climbing-stairs/solution.cpp
// 0070 · Climbing Stairs
// Approach: DP (Fibonacci) | Time: O(n) | Space: O(1)

class Solution {
public:
    int climbStairs(int n) {
        int a = 1, b = 1;
        for (int i = 1; i < n; i++) {
            int c = a + b;
            a = b;
            b = c;
        }
        return b;
    }
};

File: topics/dynamic-programming/easy/0070-climbing-stairs/solution.py
# 0070 · Climbing Stairs
# Approach: DP (Fibonacci) | Time: O(n) | Space: O(1)

class Solution:
    def climbStairs(self, n: int) -> int:
        a, b = 1, 1
        for _ in range(n - 1):
            a, b = b, a + b
        return b
